src/node/internal/internal_zlib.ts

```diff
@@ -20,6 +20,147 @@ const {
   CONST_UNZIP,
 } = zlibUtil;
 
+export function crc32(
+  data: ArrayBufferView | string,
+  value: number = 0
+): number {
+  validateUint32(value, 'value');
+  return zlibUtil.crc32(data, value);
+}
+
+export function inflateSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_INFLATE));
+}
+
+export function deflateSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATE));
+}
+
+export function gunzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_GUNZIP));
+}
+
+export function gzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(zlibUtil.zlibSync(data, options, zlibUtil.CONST_GZIP));
+}
+
+export function inflateRawSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_INFLATERAW)
+  );
+}
+
+export function deflateRawSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATERAW)
+  );
+}
+
+export function unzipSync(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {}
+): Buffer {
+  return Buffer.from(
+    zlibUtil.zlibSync(data, options, zlibUtil.CONST_DEFLATERAW)
+  );
+}
+
+export function inflate(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {},
+  callback: CompressCallback
+): void {
+  zlibUtil.zlibWithCb(data, options, zlibUtil.CONST_INFLATE, (error, result) =>
+    callback(error ? Error(error) : null, result ? Buffer.from(result) : null)
+  );
+}
+
+export function unzip(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {},
+  callback: CompressCallback
+): void {
+  zlibUtil.zlibWithCb(data, options, zlibUtil.CONST_UNZIP, (error, result) =>
+    callback(error ? Error(error) : null, result ? Buffer.from(result) : null)
+  );
+}
+
+export function inflateRaw(
+  data: ArrayBufferView | string,
+  options: ZlibOptions = {},
+  callback: CompressCallback
+): void {
+  zlibUtil.zlibWithCb(
+    data,
+    options,
+    zlibUtil.CONST_INFLATERAW,
+    (error, result) =>
+      callback(error ? Error(error) : null, result ? Buffer.from(result) : null)
```

Comment:
Just an idea:

```suggestion
      const args = error ? [error] : [null, Buffer.from(result)];
      callback(...args)
```

Reply:
It looks nicer but TS  is getting mad at me. I think I'll need your help tomorrow morning.